src/CodeXMLParse.cpp

```diff
@@ -46,8 +45,67 @@ void AnnotateOpref(ANNOTATOR_PARAMS)
 	annotation.type = R_CODE_ANNOTATION_TYPE_OFFSET;
 	annotation.offset.offset = op->getAddr().getOffset();
 }
+void AnnotateFunctionName(ANNOTATOR_PARAMS)
+{
+	const char *func_name = node.child_value();
+	if(!func_name)
+		return;
+	RCodeAnnotation annotation = {};
+	annotation.type = R_CODE_ANNOTATION_TYPE_FUNCTION_NAME;
+	pugi::xml_attribute attr = node.attribute("opref");
+	if(attr.empty())
+	{
+		if(ctx->func->getName() == func_name)
+		{
+			annotation.function_name.name = strdup(ctx->func->getName().c_str());
+			annotation.function_name.offset = ctx->func->getAddress().getOffset();
+			// Code below makes an offset annotation for the function name(for the currently decompiled function)
+			RCodeAnnotation offsetAnnotation = {};
+			offsetAnnotation.type = R_CODE_ANNOTATION_TYPE_OFFSET;
+			offsetAnnotation.offset.offset = annotation.function_name.offset;
+			out->push_back(offsetAnnotation);
+		}
+		else
+		{
+			annotation.function_name.name = strdup("INVALID: Doesn't look like a function.");
```

Comment:
Just return without adding an annotation in these cases instead of adding an invalid one. Same for the other "INVALID" ones.

Reply:
I will change this as suggested

Reply:
I have made all the suggested changes.

Reply:
I'm sorry, I misread your suggestion. I will change it now. All the other changes are as suggested.

Reply:
Done. Changed as suggested.